### PR TITLE
DPI related fixes for wxStyledTextCtrl

### PR DESCRIPTION
**Pull request by MaartenBent**
Fix and improve DPI handling in `wxStyledTextCtrl`.

Fix cursor size after DPI change.

Enable handling DPI events in `wxSTCPopupWindow` and other popup windows (that do not inherit from `wxTopLevelWindow`).

Some cleanup of STC example, and add option to select drawing technology.

### CODE REVIEW

**Comment by vadz** [comment]
Unfortunately `WM_DPICHANGED` is not defined for the old MinGW (see AppVeyor error). We probably just need to include `wx/msw/missing.h` from `src/msw/nonownedwnd.cpp` to fix it.

File: src/stc/PlatWX.cpp [hunk 1]
```diff
@@ -295,15 +297,15 @@ void SurfaceImpl::InitPixMap(int width, int height, Surface *surface, WindowID w
         hdc = new wxMemoryDC(static_cast<SurfaceImpl*>(surface)->hdc);
     else
         hdc = new wxMemoryDC();
+    ((wxMemoryDC*)hdc)->GetImpl()->SetWindow(GETWIN(winid));
```
Comment (vadz) [inline src/stc/PlatWX.cpp:300]: I'd add `wxMemory* mdc`, initialize it appropriately and then do `hdc = mdc` to avoid the ugly cast.

**Review by vadz — COMMENTED** [review]
No real comments, thanks as always for your improvements!

**Comment by MaartenBent** [comment]
> Unfortunately `WM_DPICHANGED` is not defined for the old MinGW (see AppVeyor error).

Oh, I missed that. I thought it just timed out. Should be fixed now.

**Pull request merged**